include/display.h

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2017 dXplore
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API for display drivers and applications
+ */
+
+#ifndef __DISPLAY_H__
+#define __DISPLAY_H__
+
+/**
+ * @brief Display Interface
+ * @defgroup display_interface display Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <device.h>
+#include <stddef.h>
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum display_pixel_format {
+	PIXEL_FORMAT_RGB_888 = 0,
+
+	PIXEL_FORMAT_END,
+};
+
+/**
+ * @typedef display_on_api
+ * @brief Callback API to turn display on
+ * See display_on() for argument description
+ */
+typedef int (*display_on_api)(struct device *dev);
+
+/**
+ * @typedef display_off_api
+ * @brief Callback API to turn display off
+ * See display_off() for argument description
+ */
+typedef int (*display_off_api)(struct device *dev);
+
+/**
+ * @typedef display_write_bitmap_api
+ * @brief Callback API for writing bitmap
+ * See display_write_bitmap() for argument description
+ */
+typedef int (*display_write_bitmap_api)(
+	const struct device *dev, const u16_t x, const u16_t y, const u16_t w,
+	const u16_t h, const enum display_pixel_format pixel_format,
```

Comment:
Why does  display_pixel_format have to be handed over here? What should the display driver do with it?

Reply:
This was added on request (@lpereir ):
>It would be useful to have another parameter, specifying the pixel format. Right now, the format is "whatever the display expects". Given the wide variety of displays and pixel formats, this is important.

But how more I think about it is maybe better to drop the support form the write function and leave the conversion up to a higher layer.

Reply:
Yeah, this makes sense. In fact, the suggestion to have the pixel format be a compile-time setting is the best option here -- the upper layer would perform the conversion if necessary, and the API would be easier.

Drivers can expose, as Kconfig options, the display formats it supports; maybe if a device supports more than one, you could have an API to select which one?

Reply:
Removed pixel format from write and added API function to set pixel format at run time (if supported)